### PR TITLE
fix(server): corrected saving and loading (fixes #1343)

### LINKED ISSUE #1343: "H5P saved" notification comes too early
You immediately get the "H5P saved" notification, even if the file hasn't been fully written to the disk yet. If it is a big file (several hundred MB, it won't have finished, however). This will lead to broken files if the user quits directly after saving.

### PR DESCRIPTION
**Pull request by sr258**

### CODE REVIEW

**Comment by sr258** [comment]
This should resolve some of the issues reported in Twitter.

I tried putting in a lock mechanism that completely disables user interactions (using an unclosable modal) during save and export operations, but I failed. I tried adding a `locked` property to the state of the editor, but somehow this didn't work. I still don't understand what must be done to work with Redux. @JPSchellenberg Maybe you can give this a shot, as I think it's important that users don't do something in the UI while saving and exporting is going on.

**Comment by JPSchellenberg** [comment]
Ok, I will add a locked property to the state and show a [backdrop](https://material-ui.com/components/backdrop/) while the editor is doing things.

**Comment by JPSchellenberg** [comment]
The UI is now blocked by a [backdrop](https://material-ui.com/components/backdrop/) when saving and exporting files.

**Review by JPSchellenberg — APPROVED** [review]

**Comment by sr258** [comment]
That's great!

I've played around with it and I can still do things while the backdrop is showing: I can use the keyboard to change focus behind the backdrop and execute actions. It's also still possible to use the main menu with the mouse.

Any idea how to disable that?

**Comment by JPSchellenberg** [comment]
I can disable all actions that are handled by our app like saving, exporting and creating new files. Selecting text via keyboard is handled by electron itself and disabling it would be quite some hack, so I would stay with our solution.

**Comment by sr258** [comment]
Maybe it's possible to focus the loading indicator and catch all keyboard input there?

**Comment by JPSchellenberg** [comment]
Yeah, that might be possible.. I thought you were mainly referring to things like `ctrl+a` and `ctrl+v` from the menu.

**Comment by sr258** [comment]
You can use tab to change the focus and interact with the whole app

**Comment by JPSchellenberg** [comment]
I managed to block all keyboard input while the app saves and exports h5p-files.

File: server/src/controllers/LumiController.ts [hunk 1]
```diff
@@ -28,47 +31,55 @@ export default class LumiController {
...
+                this.h5pEditor.config
+            );
+            await packageExporter.createPackage(contentId, stream, new User());
+            // We also need to wait for the stream to finish before returning, so
+            // that the user is notified correctly about fact that saving is still
+            // going on.
+            await new Promise<void>((y, n) => {
+                stream.on('finish', () => {
+                    y();
+                    electronState.setState({ blockKeyboard: false });
```
Comment (sr258) [inline server/src/controllers/LumiController.ts]: We could put the unblocking into a finally statement

**Pull request merged**